kafka-client/src/test/java/dev/responsive/kafka/integration/ResponsiveSessionStoreIntegrationTest.java

```diff
@@ -0,0 +1,263 @@
+import org.apache.kafka.streams.kstream.Windowed;
+import org.apache.kafka.streams.kstream.internals.SessionWindow;
+import org.apache.kafka.streams.state.SessionStore;
+import org.hamcrest.Matchers;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInfo;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+public class ResponsiveSessionStoreIntegrationTest {
+
+  @RegisterExtension
+  static ResponsiveExtension EXTENSION = new ResponsiveExtension(StorageBackend.MONGO_DB);
+
+  private static final String INPUT_TOPIC = "input";
+  private static final String OTHER_TOPIC = "other";
+  private static final String OUTPUT_TOPIC = "output";
+
+  private final Map<String, Object> responsiveProps = new HashMap<>();
+
+  private String name;
+  private Admin admin;
+
+  @BeforeEach
+  public void before(
+      final TestInfo info,
+      final Admin admin,
+      @ResponsiveConfigParam final Map<String, Object> responsiveProps
+  ) throws InterruptedException, ExecutionException {
+    // append a random int to avoid naming collisions on repeat runs
+    name = info.getTestMethod().orElseThrow().getName() + "-" + new Random().nextInt();
+
+    this.responsiveProps.putAll(responsiveProps);
+
+    this.admin = admin;
+    final var result = admin.createTopics(
+        List.of(
+            new NewTopic(inputTopic(), Optional.of(1), Optional.empty()),
+            new NewTopic(otherTopic(), Optional.of(1), Optional.empty()),
+            new NewTopic(outputTopic(), Optional.of(1), Optional.empty())
+        )
+    );
+    result.all().get();
+  }
+
+  @Test
+  public void shouldComputeSessionAggregate() throws Exception {
+    // Given:
+    final Duration inactivityGap = Duration.ofSeconds(5);
+    final Duration gracePeriod = Duration.ofSeconds(2);
+    final SessionWindows window =
+        SessionWindows.ofInactivityGapAndGrace(inactivityGap, gracePeriod);
+
+    final Materialized<String, String, SessionStore<Bytes, byte[]>> store =
+        ResponsiveStores.sessionMaterialized(
+            ResponsiveSessionParams.session(name, inactivityGap, gracePeriod)
+        );
+
+    // Start from timestamp of 0L to get predictable results
+    final List<KeyValueTimestamp<String, String>> inputEvents = asList(
+        new KeyValueTimestamp<>("key", "a", 0L),
+        new KeyValueTimestamp<>("key", "b", 6_000L),
+        new KeyValueTimestamp<>("key", "c", 7_000L),
+        new KeyValueTimestamp<>("key1", "d", 8_000L),
+        new KeyValueTimestamp<>("key1", "e", 16_000L),
+        new KeyValueTimestamp<>("key1", "f", 12_000L),
+        new KeyValueTimestamp<>("key1", "g", 7_500L)
+    );
+    final List<KeyValue<Windowed<String>, String>> expectedPeeks = List.of(
+        new KeyValue<>(windowedKey("key", 0, 0), "a"),
+        new KeyValue<>(windowedKey("key", 0, 0), null),
+        new KeyValue<>(windowedKey("key", 0, 6000), "ab"),
```

Comment:
I'm confused...the inactivity gap is 5000, right? So if a is at 0 and b is at 6000, then there's more than 5000ms between these two events, and therefore they should not be merged into the same session...right?

It's been a long time since I thought about session aggregations so it's completely possible I'm not looking at this the right way. But my understanding is that  events separated by more than the inactivity gap should stay separate, at least unless/until another event comes in between them and is within the inactivity gap of both. In that case you get could a and b merged with each other -- if, say, c had timestamp 3000 for example, we would see "a" and "b" merged into "abc". But I think (not 100% sure) that we'd never see just an "ab" output